google-cloud-vision/lib/google/cloud/vision/v1/helpers.rb

```diff
@@ -0,0 +1,170 @@
+            if request.features.empty?
+              VisionV1::Feature::Type.descriptor.each do |enum_value|
+                request.features.push(VisionV1::Feature.new(type: enum_value))
+              end
+            end
+
+            requests = [request]
+
+            # We only sent one request, so we can pull out the only response.
+            batch_annotate_images(requests, options: options).responses[0]
+          end
+
+          # @!macro [attach] create_feature_method
+          #   @method $1(request, options: nil)
+          #   Annotate a single image with the feature +$1+.
+          #   @param [String, IO, Hash,
+          #       Google::Cloud::Vision::V1::AnnotateImageRequest] request
+          #     The image to be annotated, specified as a
+          #     {Google::Cloud::Vision::V1::AnnotateImageRequest} or Hash of
+          #     the same form. In simple cases, you may instead specify a String
+          #     (the URL or filename of the image) or an +IO+ (the image
+          #     itself).
+          #   @param [Google::Gax::CallOptions] options
+          #     Overrides the default settings for this call, e.g, timeout,
+          #     retries, etc.
+          #   @return [Google::CLoud::Vision::V1::AnnotateImageResponse]
+          #     The annotations for the image.
+          #   @raise [Google::Gax::GaxError] if the RPC is aborted.
+          #   @raise [ArgumentError] if features other than +$1+ are requested.
+          #   @example
+          #     require "google/cloud/vision/v1"
+          #
+          #     image_annotator_client = Google::Cloud::Vision.new
+          #     response = image_annotator_client.$1("path/to/image")
+          def self.create_feature_method feature
+            define_method(feature) do |request, options: nil, **kwargs|
+              # Convert String or IO input to request object
+              request = request_to_object(request, **kwargs)
+              feature_enum = feature.upcase
+              if request.features.any? { |f| f.type = feature_enum }
+                raise ArgumentError, "Feature cannot be set explicitly"
+              end
+
+              request.features.push(
+                VisionV1::Feature.new(type: feature_enum)
+              )
+              annotate_image(request, options: options)
+            end
+          end
+
+          # Feature methods. This "loop" is unrolled so that YARD
+          # attaches documentation correctly.
+          create_feature_method(:image_properties)
+          create_feature_method(:safe_search_detection)
+          create_feature_method(:logo_detection)
+          create_feature_method(:web_detection)
+          create_feature_method(:face_detection)
+          create_feature_method(:text_detection)
+          create_feature_method(:document_text_detection)
+          create_feature_method(:landmark_detection)
+          create_feature_method(:label_detection)
+          create_feature_method(:crop_hints)
```

Comment:
I'm not getting this feature method?

```ruby
>> puts vision.methods.sort.grep(/detect/).inspect
=> [:document_text_detection, :face_detection, :label_detection, :landmark_detection, :logo_detection, :safe_search_detection, :text_detection, :web_detection]
```

The Feature.Types: https://cloud.google.com/vision/docs/reference/rpc/google.cloud.vision.v1#google.cloud.vision.v1.Feature.Type

 - ✓ FACE_DETECTION | Run face detection.
 - ✓ LANDMARK_DETECTION | Run landmark detection.
 - ✓ LOGO_DETECTION | Run logo detection.
 - ✓ LABEL_DETECTION | Run label detection.
 - ✓ TEXT_DETECTION | Run text detection / optical character recognition (OCR). Text detection is optimized for areas of text within a larger image; if the image is a document, use DOCUMENT_TEXT_DETECTION instead.
 - ✓ DOCUMENT_TEXT_DETECTION | Run dense text document OCR. Takes precedence when both DOCUMENT_TEXT_DETECTION and TEXT_DETECTION are present.
 - ✓ SAFE_SEARCH_DETECTION | Run Safe Search to detect potentially unsafe or undesirable content.
 - ❌  _missing_ IMAGE_PROPERTIES | Compute a set of image properties, such as the image's dominant colors.
 - ❌  _missing_ CROP_HINTS | Run crop hints.
 - ✓ WEB_DETECTION | Run web detection.

@geigerj

Reply:
Nevermind! They're not *_detection** methods... my bad!

Reply:
^---- I found confusing => users may find confusing (feedback for the proto, not this code 😄 )

Wrapping up samples based on this GAPIC to LGTM, wanna just test each detection method... LGTM on the way....